### DISCUSSION TITLE
Book 3, Chapter 10.2 Sampling Directions towards a Hittable - Updates to Hittable List Missing

### DISCUSSION THREAD

**Discussion by davidfrogley** [RayTracing/raytracing.github.io#1256] [Debugging]
Code listings in chapter 10.2 do not include changes to the `hittable_list` class, which should include an override of the `pdf_value` and `generate` methods from the `hittable` base class. Without these, the rendered image uses the default value for both of these (`0` for `pdf_value`, and `vec3(1,0.0)` for `generate`). This results in a completely black image, except for the light on the ceiling.

**Comment by hollasch** [comment]
For things that look like bugs or errors on our part, please create issues. I've spawned #1267 from this.

**Reply by davidfrogley** [reply]
OK, will do. Thanks.
